sched/mqueue/msgrcv.c

```diff
@@ -0,0 +1,261 @@
+/****************************************************************************
+ * sched/mqueue/msgrcv.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <assert.h>
+
+#include <nuttx/cancelpt.h>
+
+#include "sched/sched.h"
+#include "mqueue/msg.h"
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: msgrcv_wait
+ ****************************************************************************/
+
+static int msgrcv_wait(FAR struct msgq_s *msgq, FAR struct msgbuf_s **rcvmsg,
+                       long msgtyp, int msgflg)
+{
+  FAR struct list_node *newmsg = NULL;
+  FAR struct list_node *node;
```

Comment:
change the type of newmsg and node to msgbuf_s

Reply:
Done